plugins/woocommerce-admin/client/dashboard/components/connect/index.js

```diff
@@ -21,8 +21,12 @@ export class Connect extends Component {
 	}
 
 	componentDidUpdate( prevProps ) {
-		const { createNotice, error, isRequesting, onError, setIsPending } =
-			this.props;
+		const {
+			createNotice,
+			jetpackAuth: { error, isRequesting },
```

Comment:
Previously the component would figure out the URL (alongside the error and "is retrieving" status) internally. Now it receives it as a prop. This allows us to prefetch the URL which allows us to immediately present a clickable "Connect" button to users (vs. requiring them to wait until the REST API responds with a formatted Jetpack auth URL if we started retrieving that URL once the Connect button is rendered).